README.md

```diff
@@ -1,52 +1,83 @@
 # folio-graphql
-## Install dependencies
+## Getting Started
+The API is built using the [Apollo GraphQL server](https://www.apollographql.com/docs/apollo-server/), which requires nodeJS v14 or later. Install dependencies with npm:
 ```sh
 npm install
 ```
-
-## Compile typescript
+Before starting up the server, you need to configure the base URL of FOLIO's Okapi endpoint. This can be done by setting the environment variable `OKAPI_BASE_URL` or by creating a `config/local.json` file to store your configuration:
 ```sh
-npm run compile
+cp config/default.json config/local.json
 ```
-
-## Configure
-By default the configuration is set to read from environment variables `OKAPI_BASE_URL`, but you can overwrite this config by creating `config/local.json` and setting values you wish to use.
+Then, you can get a development server started quickly with:
 ```sh
-cp config/default.json config/local.json
+npm run start:dev
 ```
-
-## Start the server
+Or with `docker compose`:
 ```sh
-npm run start
+docker compose up
 ```
-The server will start on http://localhost:4000/
+## Usage
+Once the server is running, you can use the web interface to explore and build queries. In order to enable schema introspection, you need to configure Apollo to send the `okapi_username` and `okapi_password` headers with each request. You can do this by updating the connection settings at the top of the query explorer:
 
-or using docker:
+![Gear icon for connection settings in query explorer](connection_settings.png)
+## Architecture
+The GraphQL API is constructed based on FOLIO's published API schemas, in a process that involves several steps. In most cases, this pipeline will be automated for you and run when files are modified if you use the `start:dev` command. See `package.json` for commands that invoke the various steps in the pipeline separately.
+### Schema
+The schemas are generated from the RAML files published by FOLIO using a script in the `scripts` directory. The script corrects for some eccentricities in FOLIO's published JSON schemas, and then uses the [json-schema-to-graphql-types](https://github.com/lifeomic/json-schema-to-graphql-types) library to output a GraphQL schema to stdout.
+#### Adding new types
+To add a new type to the schema, you need to add it to the allowlist of enabled types in the definition for `queryType` near the bottom of the script:
+```js
+const queryType = new GraphQLObjectType({
+  name: 'Query',
+  fields: () => ({
+    // ...
+    // add new types here
+    // ...
+  })
+})
 ```
-docker compose up app
+#### Generating the GraphQL schema
+You can invoke the script with:
+```sh
+npm run update-graphql-schema
 ```
+This will output the schema to stdout. You can redirect it to a file if you want to save it:
+```sh
+npm run update-graphql-schema > schema.graphql
```

Comment:
Can you name this example something like newschema.graphql for clarity that we are not overwriting src/schema.graphql (I know you mention this below but still)

Reply:
Sure; updated